src/foundry/client-esm/applications/api/dialog.d.mts

```diff
@@ -136,20 +136,33 @@ declare class DialogV2<
    *          dialog was dismissed, and rejectClose is false, the Promise
    *          resolves to null.
    */
-  static wait<Options extends DeepPartial<DialogV2.WaitOptions>>(
-    config?: Options,
-  ): Promise<InferButtonReturnTypes<Options> | InferDismissType<Options>>;
+  static wait<Options extends DialogV2.WaitOptions>(config?: Options): Promise<DialogV2.WaitReturn<Options>>;
 
+  /**
+   * Present an asynchronous Dialog query to a specific User for response.
+   * @param user   - A User instance or a User id
+   * @param type   - The type of Dialog to present
+   * @param config - Dialog configuration forwarded on to the Dialog.prompt, Dialog.confirm, or
+   *                 Dialog.wait function depending on the query type. Callback options are not supported.
+   * @returns The query response or null if no response was provided
+   *
+   * @see {@link DialogV2.prompt}
+   * @see {@link DialogV2.confirm}
+   * @see {@link DialogV2.wait}
+   */
   static query<T extends DialogV2.Type, Options extends DeepPartial<DialogV2.QueryConfig<T>>>(
     user: User.ConfiguredInstance | string,
     type: T,
-    config: Options,
-  ): Promise<(T extends "confirm" ? boolean : string) | InferDismissType<Options>>;
+    config?: Options,
+  ): Promise<(T extends "confirm" ? boolean : string) | DialogV2.Internal.DismissType<Options>>;
```

Comment:
Can you expose the entirety of this publicly? I don't want someone to have to use the internal namespace just to subclass `query`.

How about `QueryReturnType`? Incredibly clever name I know.